src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java

```diff
@@ -102,7 +103,14 @@ public ProjectBindingManager(EnginesFactory enginesFactory, WorkspaceFoldersMana
     this.client = client;
     this.progressManager = progressManager;
     this.httpClient = httpClient;
+    this.folderBindingCache = new ConcurrentHashMap<>();
   }
+  public ProjectBindingManager(EnginesFactory enginesFactory, WorkspaceFoldersManager foldersManager, SettingsManager settingsManager, LanguageClient client,
+    ProgressManager progressManager, ApacheHttpClient httpClient, Map<URI, Optional<ProjectBindingWrapper>> folderBindingCache) {
+    this(enginesFactory, foldersManager, settingsManager, client, progressManager, httpClient);
+    this.folderBindingCache = folderBindingCache;
```

Comment:
Same comment than for AnalysisManager. You can make the code testable without removing `final` on the attribute.